### PR TITLE
prevent HeliosSoloLogService from continuing to run after tests with HeliosSoloDeployment are done

### PR DESCRIPTION
**Pull request by mattnworb**
In a test class that uses HeliosSoloDeployment as a `@Rule` (i.e. one
helios-solo instance per test method), I found that the test was hanging
for a long period of time after the test itself seemed to be finished
(judging from the class-under-test's log output).

Thread dumps showed that the thread running HeliosSoloLogService was
blocked at a call to `heliosClient.listHosts().get()`. This change adds
timeouts when using `Future.get()` to avoid any possibility of blocking
forever.

Also change the Service itself to use daemon threads with it's executor,
as we do not wish to keep the JVM running under any instance if the
HeliosSoloLogService is still running when the main thread has exited.

### CODE REVIEW

**Comment by mattnworb** [comment]
@rohansingh @spotify/helios-team

File: helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloLogService.java [hunk 1]
```diff
@@ -58,12 +64,26 @@
...
+  private static <T> T get(Future<T> future)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    return future.get(1, TimeUnit.SECONDS);
+  }
+
+  @Override
+  protected ScheduledExecutorService executor() {
+    final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+        .setNameFormat(serviceName())
+        .setDaemon(false)
```
Comment (davidxia) [inline helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloLogService.java]: Shouldn't this be set to true as daemon threads do not prevent the JVM from exiting when the program finishes.

Comment (mattnworb) [inline helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloLogService.java]: oops, that was my intent. thanks for catching

**Comment by davidxia** [comment]
LGTM

**Pull request merged**